File: tools/samples/dotnet/Domain/Basket.cs
namespace Sample.Domain;

public sealed class Basket
{
    private readonly List<decimal> lines;

    public Basket(IEnumerable<decimal> lines)
    {
        this.lines = lines.ToList();
    }

    // TODO: value added tax is not handled here yet
    public decimal Total(decimal discount)
    {
        return Net(discount);
    }

    private decimal Net(decimal discount)
    {
        var total = 0m;
        foreach (var line in lines)
        {
            total += line;
        }

        return total - discount;
    }
}

File: tools/samples/java/src/main/java/com/example/domain/Basket.java
package com.example.domain;

import com.example.services.Pricing;
import java.util.List;

public final class Basket {
    private final List<Integer> lines;

    public Basket(List<Integer> lines) {
        this.lines = List.copyOf(lines);
    }

    // TODO: value added tax is not handled here yet
    public int total(int discount) {
        return net(discount);
    }

    public String band(int discount) {
        return Pricing.band(this, discount);
    }

    private int net(int discount) {
        int sum = 0;
        for (int line : lines) {
            sum += line;
        }
        return sum - discount;
    }
}
